lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -161,6 +161,27 @@ defmodule ExUnit.Runner do
     running
   end
 
+  defp spawn_modules(
+         config,
+         [{_partition_key, partition_modules} | modules],
```

Comment:
Instead of introducing a new format, I would prefer to keep all shapes consistent, this means that we should change `sync_modules` to also be `{key, module, params}` tuples. This will work quite nicely if we chose to name the option `:group`.

Reply:
That makes sense, but it would have to be something like `{key, [{module, params}]}` (see https://github.com/elixir-lang/elixir/pull/13897#discussion_r1797641408) to support the same approach.

Then we would have a single `{module, params}` tuple list for tests which aren't grouped and a list with potentially more tuples when we use `group`.

Does that make sense?

Reply:
I've pushed https://github.com/elixir-lang/elixir/pull/13897/commits/3b9d737354a2964bd3ac188857526d9603181146 to keep the format the same for both sync and async.